pkg/devfile/generator/utils.go

```diff
@@ -34,13 +33,25 @@ func convertEnvs(vars []v1.EnvVar) []corev1.EnvVar {
 // convertPorts converts endpoint variables from the devfile structure to kubernetes ContainerPort
 func convertPorts(endpoints []v1.Endpoint) []corev1.ContainerPort {
 	containerPorts := []corev1.ContainerPort{}
+	portMap := make(map[string]bool)
 	for _, endpoint := range endpoints {
-		name := strings.TrimSpace(util.GetDNS1123Name(strings.ToLower(endpoint.Name)))
-		name = util.TruncateString(name, 15)
+		var portProtocol corev1.Protocol
+		portNumber := int32(endpoint.TargetPort)
 
+		if endpoint.Protocol == v1.UDPEndpointProtocol {
+			portProtocol = corev1.ProtocolUDP
+		} else {
+			portProtocol = corev1.ProtocolTCP
+		}
+		name := fmt.Sprintf("%d-%s", portNumber, strings.ToLower(string(portProtocol)))
+		if _, exist := portMap[name]; exist {
+			continue
+		}
```

Comment:
```suggestion
		if _, exist := portMap[name]; !exist {
	                  portMap[name] = true
	                  containerPorts = append(containerPorts, corev1.ContainerPort{
		                  Name:          name,
		                  ContainerPort: portNumber,
		                  Protocol:      portProtocol,
	                  })
                  }
```

would probably do this way but upto you

Reply:
updated